samtranslator/model/eventsources/push.py

```diff
@@ -1219,6 +1219,11 @@ def _add_auth_to_openapi_integration(self, api, editor):
                     "an explicit 'Authorizer' set or a 'DefaultAuthorizer' "
                     "configured on the HttpApi.",
                 )
+        elif not isinstance(method_authorizer, string_types):
```

Comment:
`method_authorizer` might have been updated in the preceding `if` ... do we know that `DefaultAuthorizer` is a string (was that previously checked?). If not, we could consider turning this into an `if` rather than `elif`.

Reply:
Good call, `DefaultAuthorizer` being a dict should not produce this error message (instead, it has its own message). Fixed. I also added an extra unit test to ensure only `DefaultAuthorizer` related error message is generated when it happens.